### ISSUE TITLE
Unexpected behavior for comment lines in .gitignore file

### ISSUE THREAD

**Issue by ericksoen** [bmeck/dotignore#8]
I ran into an unexpected issue using `standard-version`, which relies on this package, where a line comment in the `.gitignore` file caused an unexpected behavior.

*File System*
`package.json`

*.gitignore (auto-generated for my projects with something akin to the following
`
## TODO: If the tool you use a package structure, uncomment the next line
# !package
`

From my initial investigation, `dotignore` will modify the second line from `# !package` to ` !package` in the code block below. 
`
var commentLine = line[0] === '#'; // true
var negatedLine == line[0] === '!'; // false since the comment character is still in the initial position
if (negatedLine || commentLine || rootedLine) {
      line = line.substring(1);
}
`
Although it passes the `commentLine` boolean via the options object to the minimatch RegularExpression creator, that value is never used inside the minimatch method. Instead, the `minimatch.make` method repeats a similar check `pattern.charAt(0)` to suppress including comment lines from the regular expression set. There's some interesting interplay between the `#` character already having been removed and `negatedLine` not correctly evaluating to true--the character isn't at the zero index position in this example. At the end, the ignore pattern looks like `package`, which doesn't seem like the desired behavior for a commented line.

**Comment by ljharb** [comment]
Thanks, this seems like something worth fixing.

Would you be able to file a PR containing your failing test case?

**Comment by ericksoen** [comment]
Of course. I'll try to push something up soon. One issue I need to resolve before I can push. A fresh clone from master has a failing test (it looks like a path separator issue due to running on a Windows machine). Once I've got all the tests passing, I can submit a PR.

**Comment by ericksoen** [comment]
[Pull request submitted](https://github.com/bmeck/dotignore/pull/9)
